src/web/src/components/Posts/GitHubInfo.tsx

```diff
@@ -25,37 +26,64 @@ type Props = {
   ghUrls: string[];
 };
 
-const filterGitHubUrls = (ghUrls: string[]) => {
+const filterGitHubUrls = (urls: string[]) => {
   const issues: string[] = [];
   const pullRequests: string[] = [];
   const repos: string[] = [];
+  const commits: string[] = [];
 
-  ghUrls.forEach((ghUrl) => {
-    // add unique repo urls to repos array
-    if (
-      ghUrl.match(/https:\/\/github\.com\/[^\/]+\/[^\/]+/) &&
-      !repos.includes(ghUrl.replace(/(https:\/\/github\.com\/[^\/]+\/[^\/]+)\/.*/, '$1'))
-    ) {
-      repos.push(ghUrl.replace(/(https:\/\/github\.com\/[^\/]+\/[^\/]+)\/.*/, '$1'));
-    }
+  const ghUrls = urls.map(parseGitHubUrl).filter((url) => url !== null) as URL[];
```

Comment:
This style of using `.map()` is frowned upon, since the call signature of the function really has multiple arguments, even though you're only using the first:

```js
// map(function callbackFn(element, index, array) { ... }, thisArg)
const ghUrls = urls.map(url => parseGitHubUrl(url)).filter((url) => url !== null) as URL[];
```